S3Client/S3Client.cs

```diff
@@ -112,7 +111,22 @@ public void Dispose()
 
     private string GetBaseUrl()
     {
-      return _config.CustomDomain ?? _config.Endpoint;
+      string url = _config.CustomDomain ?? _config.Endpoint;
+
+      // I was considering removing the url setting above entirely, as it doesn't seem to work. But it may just be my problem, so if it fails, continue to setting it through getting the bucket
```

Comment:
I believe you may have addressed it further down, as I saw when I skimmed, but the exact issue I had was that absolutely nothing was being put into the "url" string, which is why my code checked for the string being empty or null.

Reply:
All here makes sense to me, but I also don't think we should expect an end user to know whether a bucket region has been launched before a specific date or not, it might be best we default it to us-east-1 so that it is more clear that they need to change it should they be using a different endpoint, my recent commit has done this. Could also mark it as placeholder text, maybe.